Automattic/wp-calypso
# Domains: Add inbound transfer status check to new transfer flow

#### Changes proposed in this Pull Request
Related to #56554. 
Adds the status check to prevent domains that are not yet available to transfer from being used on the new transfer flow.
Also moves some of the checks around so that we make a single API call instead of two.

#### Preview (for a domain that is not yet available for transfers)
![image](https://user-images.githubusercontent.com/18705930/135137195-dc5743b7-75c7-4513-bf72-b834a94c8a80.png)

#### Testing instructions
- Visit the "Use a domain I own" option from the `domains/manage` page and add the `?flags=domains/new-transfer-flow` query param to enable the new transfer flow feature flag;
- Check that the "Transfer" button is:
  - Disabled, if you try to use a domain that is not yet available for transfers;
  - Enabled, if you try to use a domain that is already available for transfers.
